Automattic/wp-calypso
# State: Add data layer and action creator for saving Jetpack Onboarding

This PR introduces an action creator and data layer for saving Jetpack Onboarding settings.

To test:
* Checkout this branch on your local Calypso.
* Make sure your Jetpack site has the latest Jetpack master.
* Apply D7342-code on your wp.com sandbox and make sure you're sandboxing the API and jetpack.wordpress.com.
* Go to https://YourJetpackSandbox.com/wp-admin/admin.php?page=jetpack&action=onboard&calypso_env=development where `YourJetpackSandbox.com` is the domain of your Jetpack sandbox.
* Wait until being redirected to the JPO flow in Calypso.
* In your console, type `state.jetpackOnboarding.credentials` to see the ID of the site you're onboarding.
* In your console, type: `dispatch( { type: 'JETPACK_ONBOARDING_SETTINGS_SAVE', siteId: 1234, settings: { siteTitle: 'New Title Here' } } ) ` (where `siteId` contains the ID of the site from the previous step, and `siteTitle` - the new title of your site).
* Go to your Jetpack site and verify title has been updated properly.
* Verify all JPO state tests pass: `npm run test-client jetpack-onboarding`
